tests/test_bpa_register.py

```diff
@@ -0,0 +1,220 @@
+        "email": "invalid-email",
+        "organizations": {},
+    }
+
+    response = client_with_settings_override.post("/bpa/register", json=invalid_data)
+
+    assert response.status_code == 422
+
+
+def test_no_selected_organizations(
+    client_with_settings_override, mock_auth_token, mocker
+):
+    """Test registration with no organizations selected"""
+    data = VALID_REGISTRATION_DATA.copy()
+    data["organizations"] = {
+        "bpa-bioinformatics-workshop": False,
+        "cipps": False,
+        "ausarg": False,
+    }
+
+    mock_response = MagicMock()
+    mock_response.status_code = 201
+    mock_response.json.return_value = {"user_id": "auth0|123"}
+
+    mock_post = mocker.patch("httpx.AsyncClient.post", return_value=mock_response)
+
+    response = client_with_settings_override.post("/bpa/register", json=data)
+
+    assert response.status_code == 200
+    called_data = mock_post.call_args[1]["json"]
+    bpa_service = called_data["app_metadata"]["services"][0]
+    assert len(bpa_service["resources"]) == 0
+
+
+def test_empty_organizations_dict(
+    client_with_settings_override, mock_auth_token, mocker
+):
+    """Test registration with empty organizations dictionary"""
+    data = VALID_REGISTRATION_DATA.copy()
+    data["organizations"] = {}
+
+    mock_response = MagicMock()
+    mock_response.status_code = 201
+    mock_response.json.return_value = {"user_id": "auth0|123"}
+
+    mock_post = mocker.patch("httpx.AsyncClient.post", return_value=mock_response)
+
+    response = client_with_settings_override.post("/bpa/register", json=data)
+
+    assert response.status_code == 200
+    called_data = mock_post.call_args[1]["json"]
+    bpa_service = called_data["app_metadata"]["services"][0]
+    assert len(bpa_service["resources"]) == 0
+
+
+def test_registration_email_format(client_with_settings_override):
+    """Test email format validation"""
+    data = VALID_REGISTRATION_DATA.copy()
+    data["email"] = "invalid-email"
+
+    response = client_with_settings_override.post("/bpa/register", json=data)
+
+    assert response.status_code == 422
+    assert "email" in response.json()["detail"][0]["loc"]
+
+
+def test_all_organizations_selected(
+    client_with_settings_override, mock_auth_token, mocker
+):
+    """Test registration with all organizations selected"""
+    data = VALID_REGISTRATION_DATA.copy()
+    data["organizations"] = {org["id"]: True for org in Settings().organizations}
+
+    mock_response = MagicMock()
+    mock_response.status_code = 201
+    mock_response.json.return_value = {"user_id": "auth0|123"}
+
+    mock_post = mocker.patch("httpx.AsyncClient.post", return_value=mock_response)
+
+    response = client_with_settings_override.post("/bpa/register", json=data)
+
+    assert response.status_code == 200
+    called_data = mock_post.call_args[1]["json"]
+    bpa_service = called_data["app_metadata"]["services"][0]
+    assert len(bpa_service["resources"]) == len(Settings().organizations)
```

Comment:
Think this test fails because you try to use `Settings()` directly, which tries to read from the `.env` file. You can use the `mock_settings` fixture instead (although you might have to check if it uses the default list of orgs).